relay-aws-extension/src/aws_extension.rs

```diff
@@ -0,0 +1,193 @@
+use actix::actors::signal::{Signal, SignalType};
+use actix::prelude::*;
+use failure::Fail;
+use relay_system::Controller;
+use reqwest::blocking::Client;
+use reqwest::StatusCode;
+use serde::Deserialize;
+use std::collections::HashMap;
+
+const EXTENSION_NAME: &str = "sentry-lambda-extension";
+const EXTENSION_NAME_HEADER: &str = "Lambda-Extension-Name";
+const EXTENSION_ID_HEADER: &str = "Lambda-Extension-Identifier";
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+struct RegisterResponse {
+    pub function_name: String,
+    pub function_version: String,
+    pub handler: String,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+struct Tracing {
+    pub r#type: String,
+    pub value: String,
+}
+
+#[derive(Debug, Deserialize)]
+#[allow(dead_code)]
+#[serde(rename_all = "UPPERCASE", tag = "eventType")]
+enum NextEventResponse {
+    #[serde(rename_all = "camelCase")]
+    Invoke {
+        deadline_ms: u64,
+        request_id: String,
+        invoked_function_arn: String,
+        tracing: Tracing,
+    },
+    #[serde(rename_all = "camelCase")]
+    Shutdown {
+        shutdown_reason: String,
+        deadline_ms: u64,
+    },
+}
+
+#[derive(Debug, Fail)]
+#[fail(display = "aws extension error")]
+/// Generic error in an AWS extension context
+pub struct AwsExtensionError;
+
+/// Actor implementing an AWS extension
```

Comment:
Maybe put a link to the AWS Extensions API reference here?

-> https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html#runtimes-extensions-registration-api

Reply:
or the whole Lambda Extesnsions API documentation:

https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html#runtimes-extensions-registration-api

(the link above is just a anchor in this link)